cpp/lib/src/LocalDataDB.cc

```diff
@@ -57,8 +65,10 @@ LocalDataDB::~LocalDataDB() {
 
 
 void LocalDataDB::clear() {
-    db_connection_->queryOrDie("DROP TABLE IF EXISTS local_data");
     db_connection_->queryOrDie("DROP TABLE IF EXISTS local_ppns_to_title_ppns_map");
+    db_connection_->queryOrDie("DROP TABLE IF EXISTS local_data");
```

Comment:
Is this really "clear()"? I would suggest that clear would use TRUNCATE TABLE, but if you drop+recreate all tables its more like RESET or sth.

Reply:
It is a clear() as it clears out the tables.  That I drop and re-create the tables is an implementation detail that the caller does neither know, nor care about.  See also https://stackoverflow.com/questions/4280041/truncate-a-sqlite-table-if-it-exists